mlflow/gateway/utils.py

```diff
@@ -80,3 +83,14 @@ def get_gateway_uri() -> str:
             f"`mlflow.set_gateway_uri()` or set the environment variable {MLFLOW_GATEWAY_URI.name} "
             "to the running Gateway API server's uri"
         )
+
+
+def assemble_uri_path(paths: List[str]) -> str:
+    """
+    Assemble a correct URI path from a list of path parts.
+
+    :param paths: A list of strings representing parts of a URI path.
+    :return: A string representing the complete assembled URI path.
+    """
+    stripped_paths = [path.strip("/") for path in paths if path]
+    return "/" + posixpath.join(*stripped_paths)
```

Comment:
If one of the `paths` has a leading slash, we could end up with two slashes here, right? Can we do an `lstrip("/")` on each path too?

Reply:
added